### PR TITLE
fix cannot use this inside model

### PR DESCRIPTION
**Pull request by thakerng**
Cannot use this inside model because in Promise.try haven't passed context of model yet.

### CODE REVIEW

**Comment by maxtruxa** [comment]
I think it wasn't the intention to allow this (not pun intended) as `model` was supposed to be just a plain object, but I like the change. The ability to use `this` would be really useful to implement a stateful model.

File: lib/grant-types/abstract-grant-type.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ function AbstractGrantType(options) {
 
 AbstractGrantType.prototype.generateAccessToken = function() {
   if (this.model.generateAccessToken) {
-    return Promise.try(this.model.generateAccessToken);
+    return Promise.try(this.model.generateAccessToken,[], this.model);
```
Comment (nunofgs) [inline lib/grant-types/abstract-grant-type.js]: Can you add a space before `[]` please? 😄

File: lib/grant-types/abstract-grant-type.js [hunk 2]
```diff
@@ -48,7 +48,7 @@ AbstractGrantType.prototype.generateAccessToken = function() {
 
 AbstractGrantType.prototype.generateRefreshToken = function() {
   if (this.model.generateRefreshToken) {
-    return Promise.try(this.model.generateRefreshToken);
+    return Promise.try(this.model.generateRefreshToken,[], this.model);
```
Comment (nunofgs) [inline lib/grant-types/abstract-grant-type.js]: Can you add a space before `[]` please? 😄

**Comment by mjsalinger** [comment]
Closing due to inactivity.

**Pull request closed without merging**